### ISSUE TITLE
[ERROR] (Infreemation::ResponseError) "JSON invalid"

### ISSUE THREAD

**Issue by garethrees** [mysociety/foi-for-councils#149]
Similar to https://github.com/mysociety/foi-for-councils/issues/122, but making a new issue for this as the cause looks to be slightly different.

```
An Infreemation::ResponseError occurred in background at 2018-06-26 11:14:35 UTC :

JSON invalid
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/infreemation-0.1.0/lib/infreemation/api.rb:50:in `rescue in block in parser'

-------------------------------
Backtrace:
-------------------------------

/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/infreemation-0.1.0/lib/infreemation/api.rb:50:in `rescue in block in parser'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/infreemation-0.1.0/lib/infreemation/api.rb:45:in `block in parser'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/rest-client-2.0.2/lib/restclient/request.rb:807:in `process_result'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/rest-client-2.0.2/lib/restclient/request.rb:725:in `block in transmit'
/opt/rbenv/versions/2.5.1/lib/ruby/2.5.0/net/http.rb:910:in `start'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/rest-client-2.0.2/lib/restclient/request.rb:715:in `transmit'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/rest-client-2.0.2/lib/restclient/request.rb:145:in `execute'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/rest-client-2.0.2/lib/restclient/request.rb:52:in `execute'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/rest-client-2.0.2/lib/restclient/resource.rb:67:in `post'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/infreemation-0.1.0/lib/infreemation/api.rb:17:in `get'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/infreemation-0.1.0/lib/infreemation/request.rb:19:in `where'
/data/vhost/hackney-foi.mysociety.org/foi-for-councils-2018-06-21T15-03-35/app/services/disclosure_log.rb:15:in `import'
/data/vhost/hackney-foi.mysociety.org/foi-for-councils-2018-06-21T15-03-35/app/workers/import_disclosure_log_worker.rb:11:in `perform'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:187:in `execute_job'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:169:in `block (2 levels) in process'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/middleware/chain.rb:128:in `block in invoke'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/middleware/chain.rb:133:in `invoke'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:168:in `block in process'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:139:in `block (6 levels) in dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/job_retry.rb:98:in `local'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:138:in `block (5 levels) in dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/rails.rb:42:in `block in call'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.0/lib/active_support/execution_wrapper.rb:87:in `wrap'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.0/lib/active_support/reloader.rb:73:in `block in wrap'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.0/lib/active_support/execution_wrapper.rb:87:in `wrap'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.0/lib/active_support/reloader.rb:72:in `wrap'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/rails.rb:41:in `call'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:134:in `block (4 levels) in dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:199:in `stats'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:129:in `block (3 levels) in dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/job_logger.rb:8:in `call'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:128:in `block (2 levels) in dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/job_retry.rb:73:in `global'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:127:in `block in dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/logging.rb:48:in `with_context'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/logging.rb:42:in `with_job_hash_context'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:126:in `dispatch'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:167:in `process'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:85:in `process_one'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/processor.rb:73:in `run'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/util.rb:16:in `watchdog'
/data/vhost/hackney-foi.mysociety.org/shared/vendor/bundle/ruby/2.5.0/gems/sidekiq-5.1.3/lib/sidekiq/util.rb:25:in `block in safe_thread'

-------------------------------
Data:
-------------------------------

* data: {:sidekiq=>
{:context=>"Job raised exception",
:job=>
{"class"=>"ImportDisclosureLogWorker",
"args"=>["week"],
"queue"=>"default",
"retry"=>true,
"jid"=>"f0e612aee3366203073223df",
"created_at"=>1530011031.224796,
"enqueued_at"=>1530011674.5789468,
"error_message"=>"JSON invalid",
"error_class"=>"Infreemation::ResponseError",
"failed_at"=>1530011031.7535148,
"retry_count"=>4,
"retried_at"=>1530011384.5326269},
:jobstr=>
"{\"class\":\"ImportDisclosureLogWorker\",\"args\":[\"week\"],\"queue\":\"default\",\"retry\":true,\"jid\":\"f0e612aee3366203073223df\",\"created_at\":1530011031.224796,\"enqueued_at\":1530011674.5789468,\"error_message\":\"JSON invalid\",\"error_class\":\"Infreemation::ResponseError
```

This is happening because we're getting invalid JSON back from Infreemation. You can inspect a response like:

```ruby
Infreemation::Request.where(rt: 'published', startdate: '2018-06-01', enddate: '2018-06-30')
```

The problem is that they're not escaping or encoding HTML in the response:

```
"responsebody": "<p><span style="font-size:12px"><span style="font-fa...
```

**Comment by garethrees** [comment]
[Emailed Infreemation](https://groups.google.com/a/mysociety.org/forum/#!topic/clientsupport-council-foi/gag_H1bhL-s) about this.

**Comment by garethrees** [comment]
Looks like they've [fixed the feed](https://groups.google.com/a/mysociety.org/d/msg/clientsupport-council-foi/gag_H1bhL-s/hfTeb3esCAAJ).

I've tested the above and we can at least parse the JSON.

Need to check what happens when we import escaped an encoded text, and figure out how to import nightly.

**Comment by garethrees** [comment]
The feed looks good, and #150 has been merged so that escaped/encoded results can be displayed.

#152 moves the import to outside of business hours.

I've [noted](https://groups.google.com/a/mysociety.org/d/msg/clientsupport-council-foi/gag_H1bhL-s/1UnyaP4JDgAJ) that the items in the feed don't contain the `datepublished` attribute, which may affect https://github.com/mysociety/foi-for-councils/issues/141, but shouldn't be a problem for closing this issue.

**Comment by garethrees** [comment]
Marking this as "awaiting deploy" because we need https://github.com/mysociety/foi-for-councils/pull/150 deployed before we can re-enable the feed.

We should also deploy https://github.com/mysociety/foi-for-councils/issues/152 so that we're not hammering their API when turning it back on.

**Comment by garethrees** [comment]
I've re-enabled the feed https://groups.google.com/a/mysociety.org/d/msg/clientsupport-council-foi/gag_H1bhL-s/6_VB1Q6TBwAJ
